meilisearch-release-changelog/v1.11.0.md

```diff
@@ -0,0 +1,177 @@
+
+# New features and updates 🔥
+
+## Experimental - Hybrid search improvements
+
+For a future stabilization, multiple changes have been applied to hybrid search features.
+
+📖 More details about the usage in this [page](https://www.notion.so/meilisearch/v1-11-AI-search-changes-0e37727193884a70999f254fa953ce6e).
+
+Done by @dureuill in #4906, #4920, #4892, and #4938.
+
+### ⚠️ Breaking changes
+
+* When using the semantic or the hybrid search, `hybrid.embedder` is now a **mandatory** parameter in `GET and POST /indexes/{:indexUid}/search`
+* As a consequence, it is now **mandatory** to pass `hybrid` even for full-vector search (with only `vector` and not `q`)
+* `embedder` is now a **mandatory** parameter in `GET and POST /indexes/{:indexUid}/similar`
+* Ignore non-zero `semanticRatio` when `vector` is passed but not `q`: a semantic search will be performed (unless `hybrid.semanticRatio` is null).
+
+### Addition & improvements
+
+* The default model for OpenAI is now `text-embedding-3-small` instead of `text-embedding-ada-002`.
+* Limit the maximum length of a rendered document template: when the source of an embedder is set to `huggingFace`, `openAi`, `rest` or `ollama`, then `documentTemplateMaxBytes` is now available as an optional parameter. This parameter describes the number of bytes in which the rendered document template text should fit when trying to embed a document. Longer texts are truncated to fit.
+* Add the ability to query whether a field is searchable in `documentTemplate`: use `field.is_searchable: true` if the field is a searchable attribute, otherwise false.
+* The new document template changed to
+  * filter out fields that are not searchable
+  * filter out fields that are nil or missing for the current document
+
+v1.10 and before:
+```
+{% for field in fields %}
+{{ field.name }}: {{ field.value }}\n
+{% endfor %}
+```
+
+From v1.11:
+```
+{% for field in fields %}
+{% if field.is_searchable and not field.value == nil %}
+{{ field.name }}: {{ field.value }}\n
+{% endif %}
+{% endfor %}
+```
+
+This last change is backward compatible. Embedders that use the v1.10 document template will keep using it. The new default document template will only be applied to newly added embedders.
+
+## Speed up indexing of your vector database with binary quantization
+
+TBD
+
+## Federated search improvements
+
+### Apply facets
+
+Apply facet in your federated search, by using `federation.facetsByIndex` in the POST `POST /multi-search` route.
+
+```json5
+POST /multi-search
+{
+  "federation": {
+    "limit": 20,
+    "offset": 0,
+	  "facetsByIndex": {
+	    "movies": ["title", "id"],
+	    "comics": ["title"],
+	  }
+  },
+  "queries": [
+    {
+      "q": "Batman",
+      "indexUid": "movies"
+    },
+    {
+      "q": "Batman",
+      "indexUid": "comics"
+    }
+  ]
+}
+```
+
```

Comment:
Could we add an example of response?


````suggestion
```json5
{
  "hits": [
    {
      "id": 42,
      "title": "Batman returns",
      "overview": "..", 
      "_federation": {
        "indexUid": "movies",
        "queriesPosition": 0
      }
    },
    {
      "comicsId": "batman-killing-joke",
      "description": "..",
      "title": "Batman: the killing joke",
      "_federation": {
        "indexUid": "comics",
        "queriesPosition": 1
      }
    },
    ..
  ],
  "processingTimeMs": 0,
  "limit": 20,
  "offset": 0,
  "estimatedTotalHits": 2,
  "semanticHitCount": 0,
  // ✨NEW✨
  // Object containing one key per index appearing in `facets`
  "facetsByIndex": {
    "movies": {
      // ✨NEW✨
      // The facet distribution for queries on index `movies`
      "distribution": {
        "title": {
          "Batman returns": 1
        },
        "id": {
          "42": 1
        }
      },
      // ✨NEW✨
      // The facet stats for queries on index `movies`
      "stats": {
        "id": {
          "min": 42,
          "max": 42
        }
      }
    },
    "comics": {
      "distribution": {
        "title": {
          "Batman: the killing joke": 1
        }
      },
      "stats": {}
    }
  }
}
```

````

Reply:
I personnaly think it makes the changelog heavier, especially because it's not really "new". But let's make @guimachiavelli decide when he will review 😊